cpplint/cpplint.py

```diff
@@ -536,10 +548,28 @@
 # This is set by --extensions flag.
 _valid_extensions = set(['cc', 'h', 'cpp', 'cu', 'cuh'])
 
+# Treat all headers starting with 'h' equally: .h, .hpp, .hxx etc.
+# This is set by --headers flag.
+_hpp_headers = None
```

Comment:
You'll want to have a default value for this instead of None. Then you be able to shorten IsHeaderExtension() to juste "file_extension in _hpp_headers".